core/modules/widgets/button.js

```diff
@@ -219,14 +207,33 @@ ButtonWidget.prototype.execute = function() {
 	this.makeChildWidgets();
 };
 
+ButtonWidget.prototype.assignDomNodeClasses = function() {
+	var classes = this.getAttribute("class","").split(" "),
+		isPoppedUp = (this.popup || this.popupTitle) && this.isPoppedUp();
+	if(this.selectedClass) {
+		if((this.set || this.setTitle) && this.setTo && this.isSelected()) {
+			$tw.utils.pushTop(classes,this.selectedClass.split(" "));
+		}
+		if(isPoppedUp) {
+			$tw.utils.pushTop(classes,this.selectedClass.split(" "));
+		}
+	}
+	if(isPoppedUp) {
+		$tw.utils.pushTop(classes,"tc-popup-handle");
+	}
+	this.domNode.className = classes.join(" ");
+}
+
 /*
 Selectively refreshes the widget if needed. Returns true if the widget or any of its children needed re-rendering
 */
 ButtonWidget.prototype.refresh = function(changedTiddlers) {
 	var changedAttributes = this.computeAttributes();
-	if(changedAttributes.actions || changedAttributes.to || changedAttributes.message || changedAttributes.param || changedAttributes.set || changedAttributes.setTo || changedAttributes.popup || changedAttributes.hover || changedAttributes["class"] || changedAttributes.selectedClass || changedAttributes.style || changedAttributes.dragFilter || changedAttributes.dragTiddler || (this.set && changedTiddlers[this.set]) || (this.popup && changedTiddlers[this.popup]) || (this.popupTitle && changedTiddlers[this.popupTitle]) || changedAttributes.setTitle || changedAttributes.setField || changedAttributes.setIndex || changedAttributes.popupTitle || changedAttributes.disabled) {
+	if(changedAttributes.actions || changedAttributes.to || changedAttributes.message || changedAttributes.param || changedAttributes.set || changedAttributes.setTo || changedAttributes.popup || changedAttributes.hover || changedAttributes.selectedClass || changedAttributes.style || changedAttributes.dragFilter || changedAttributes.dragTiddler || (this.set && changedTiddlers[this.set]) || (this.popup && changedTiddlers[this.popup]) || (this.popupTitle && changedTiddlers[this.popupTitle]) || changedAttributes.setTitle || changedAttributes.setField || changedAttributes.setIndex || changedAttributes.popupTitle || changedAttributes.disabled) {
```

Comment:
` changedAttributes.selectedClass` will always be "undefined" since `this.computeAttributes()` uses the name `class`

Reply:
A lot of other widgets have the same problem. eg: radio and chackbox. ... I'm creating PRs atm.

Reply:
I also think constructions like: `(this.set && changedTiddlers[this.set])` should not be here and we should use better init values in the `.execute()` function, to avoid the `&&`

Reply:
> ` changedAttributes.selectedClass` will always be "undefined" since `this.computeAttributes()` uses the name `class`

You might be confusing `selectedClass` with `class`, they are two distinct widget attributes.

I've targeted these specific widgets because the cost of re-rendering them can be a lot higher as they can contain a lot of nested widgets (e.g when used with tag="div" for instance").

Reply:
uups. I didn't expand the code, so I thought it's an issue.